extensions/wikia/PhalanxII/PhalanxSpecialController.class.php

```diff
@@ -370,33 +364,19 @@ public function validate() {
 		}
 
 		if ( $token == $this->getToken() ) {
-			$this->setVal( 'valid', $this->service->validate( $regexp ) );
+			try {
+				$phalanxService = PhalanxServiceFactory::getServiceInstance();
+				$this->setVal( 'valid', $phalanxService->doRegexValidation( $regexp ) );
+			} catch ( PhalanxServiceException $phalanxServiceException ) {
+				$this->setVal( 'valid', true );
+			} catch ( RegexValidationException $regexValidationException ) {
+				$this->setVal( 'valid', $regexValidationException->getMessage() );
+			}
 		}
 
 		wfProfileOut( __METHOD__ );
 	}
 
-	public function matchBlock() {
-		if ( !$this->userCanExecute( $this->wg->User ) ) {
-			$this->displayRestrictionError();
-			return;
-		}
-
-		$result = array();
-		$token = $this->request->getVal( 'token' );
-		$block = $this->request->getVal( 'block' );
-
-		if ( $token == $this->getToken() ) {
-			foreach ( Phalanx::getSupportedTypeNames() as $type => $typeName ) {
-				$blocks = $this->service->match( $type, $block );
-				if ( !empty( $blocks ) ) {
-					$result[$type] = $blocks;
-				}
-			}
-		}
-		$this->setVal( 'blocks', $result );
-	}
-
 	private function getToken() {
 		return $this->wg->User->getEditToken();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app